src/components/Grid/index.d.ts

```diff
@@ -0,0 +1,9 @@
+import GridRow from "./GridRow";
+import GridCol from "./GridCol";
+
+const Grid = {
```

Comment:
Not sure if this will work as expected. It should complain with
```
A 'declare' modifier is required for a top level declaration in a .d.ts file.
```
Maybe use the same pattern as the rest but instead let this one only have the `NestedComponents` interface. Don't extend any react type so that it doesn't get mistaken as a component `<Grid />`